src/__test__/table-theme-colors.spec.js

```diff
@@ -0,0 +1,481 @@
+          isBackgroundTransparentColor: true,
+        });
+      });
+    });
+  });
+
+  describe('when there is background color on table', () => {
+    describe('when there is only background color on table', () => {
+      it('should return defaultValue, backgroundColor, and tableBackgroundColorFromTheme when the background color on table is light and opaque', () => {
+        spyGetStyle.mockReturnValue('#fff');
+        let result = tableThemeColors(theme);
+        expect(result).toEqual({ ...defaultValue, backgroundColor: '#fff', tableBackgroundColorFromTheme: '#fff' });
+
+        spyGetStyle.mockReturnValue('rgba(255, 255, 255, 0.2)');
+        result = tableThemeColors(theme);
+        expect(result).toEqual({
+          ...defaultValue,
+          backgroundColor: 'rgba(255, 255, 255, 0.2)',
+          tableBackgroundColorFromTheme: 'rgba(255, 255, 255, 0.2)',
+        });
+      });
+
+      it('should return the darkBackgroundColorValue, backgroundColor, and tableBackgroundColorFromTheme when the background color on table is dark and opaque', () => {
+        spyGetStyle.mockReturnValue('#000');
+        let result = tableThemeColors(theme);
+        expect(result).toEqual({
+          ...darkBackgroundColorValue,
+          backgroundColor: '#000',
+          tableBackgroundColorFromTheme: '#000',
+        });
+
+        spyGetStyle.mockReturnValue('rgba(0, 0, 0, 0.2)');
+        result = tableThemeColors(theme);
+        expect(result).toEqual({
+          ...darkBackgroundColorValue,
+          backgroundColor: 'rgba(0, 0, 0, 0.2)',
+          tableBackgroundColorFromTheme: 'rgba(0, 0, 0, 0.2)',
+        });
+      });
+
+      it('should return the defaultValue, backgroundColor, and tableBackgroundColorFromTheme, isBackgroundTransparentColor to be true when the background color on table is transparent', () => {
+        spyGetStyle.mockReturnValue('rgba(0, 0, 0, 0)');
+        let result = tableThemeColors(theme);
+        expect(result).toEqual({
+          ...defaultValue,
+          backgroundColor: 'rgba(0, 0, 0, 0)',
+          tableBackgroundColorFromTheme: 'rgba(0, 0, 0, 0)',
+          isBackgroundTransparentColor: true,
+        });
+
+        spyGetStyle.mockReturnValue('rgba(255, 255, 255, 0)');
+        result = tableThemeColors(theme);
+        expect(result).toEqual({
+          ...defaultValue,
+          backgroundColor: 'rgba(255, 255, 255, 0)',
+          tableBackgroundColorFromTheme: 'rgba(255, 255, 255, 0)',
+          isBackgroundTransparentColor: true,
+        });
+
+        spyGetStyle.mockReturnValue('#11111100');
+        result = tableThemeColors(theme);
+        expect(result).toEqual({
+          ...defaultValue,
+          backgroundColor: '#11111100',
+          tableBackgroundColorFromTheme: '#11111100',
+          isBackgroundTransparentColor: true,
+        });
+      });
+    });
+
+    describe('when there is background color on table and background color on object', () => {
```

Comment:
I think you can scratch all of these. This should behave exactly as the prev describe, since it always picks the table color instead of the object, and the object color has no affect, right?

Reply:
Right, fixed those